### ISSUE TITLE
Segmentation fault (core dumped)

### ISSUE THREAD

**Issue by zhangzhiyuan303** [tile-ai/tilelang-ascend#61]
I referenced gemm_aot and tried to debug sparse_flash_attention, but I encountered an error: init successful!  **Segmentation fault (core dumped)**

This is the function declaration which I generated using [print(func.get_kernel_source())].
1.CATLASS_GLOBAL
void main_kernel( GM_ADDR Q_handle,  GM_ADDR KV_handle,  GM_ADDR Indices_handle,  GM_ADDR Output_handle,  GM_ADDR workspace_1_handle,  GM_ADDR workspace_2_handle,  GM_ADDR workspace_3_handle,  GM_ADDR workspace_4_handle,  GM_ADDR workspace_5_handle, uint64_t fftsAddr) {

2.Here is the call script
B, S, SKV, H, HKV, DQK, DV, topk = 1, 128, 32768, 128, 1, 576, 512, 2048
dtype = torch.float16

KV_stride = 1
q_start_s_index = 4096 * 7

q = torch.randn((B, S, H, DQK), dtype=dtype)
kv = torch.randn((B, SKV, HKV, DQK), dtype=dtype)
indices = torch.full((B, S, HKV, topk), SKV, dtype=torch.int32)
for b in range(B):
    for t in range(S):
        for h in range(HKV):
            i_i = torch.randperm(max(1, ((t + q_start_s_index) // KV_stride)))[:topk]
            indices[b, t, h, :len(i_i)] = i_i
output = torch.empty((B, S, H, DV), dtype=dtype)
workspace_1 = torch.zeros((256, 64, 512), dtype=dtype)
workspace_2 = torch.zeros((256, 64, 64), dtype=dtype)
workspace_3 = torch.zeros((256, 64, 64), dtype=torch.float)
workspace_4 = torch.zeros((256, 64, 64), dtype=dtype)
workspace_5 = torch.zeros((256, 64, 512), dtype=torch.float)

dump_workspace = torch.empty(75 * 1024 * 1024 // 4).float().npu()
print("init successful!")

lib_path = "./kernel_lib.so"
lib = ctypes.CDLL(lib_path)

stream = torch.npu.current_stream()._as_parameter_

def tl_sfa():
    return lib.call(
        ctypes.c_void_p(q.data_ptr()),
        ctypes.c_void_p(kv.data_ptr()),
        ctypes.c_void_p(indices.data_ptr()),
        ctypes.c_void_p(output.data_ptr()),        
        ctypes.c_void_p(workspace_1.data_ptr()),
        ctypes.c_void_p(workspace_2.data_ptr()),
        ctypes.c_void_p(workspace_3.data_ptr()),
        ctypes.c_void_p(workspace_4.data_ptr()),
        ctypes.c_void_p(workspace_5.data_ptr()),
        ctypes.c_void_p(dump_workspace.data_ptr()),
        stream)

**tl_sfa()**
dump_workspace.detach().cpu().numpy().tofile("dump_workspace.bin")
torch.npu.synchronize()

**Comment by hedi515** [comment]
With reference to the parameters of the **void call** function in the compiled **.cpp** file, you must ensure that the arguments passed to **lib.call** in your code correspond exactly. A mismatch in the number or type of parameters will result in an error. Your code should remove **ctypes.c_void_p(dump_workspace.data_ptr())** and be modified as follows:

##dump_workspace = torch.empty(75 * 1024 * 1024 // 4).float().npu()
torch.npu.synchronize()
print("init successful!")

lib_path = "./kernel_lib.so"
lib = ctypes.CDLL(lib_path)

stream = torch.npu.current_stream()._as_parameter_

def tl_sfa():
    return lib.call(
        ctypes.c_void_p(q.data_ptr()),
        ctypes.c_void_p(kv.data_ptr()),
        ctypes.c_void_p(indices.data_ptr()),
        ctypes.c_void_p(output.data_ptr()),
        ctypes.c_void_p(workspace_1.data_ptr()),
        ctypes.c_void_p(workspace_2.data_ptr()),
        ctypes.c_void_p(workspace_3.data_ptr()),
        ctypes.c_void_p(workspace_4.data_ptr()),
        ctypes.c_void_p(workspace_5.data_ptr()),
        ##ctypes.c_void_p(dump_workspace.data_ptr()),
        stream)

tl_sfa()
##dump_workspace.detach().cpu().numpy().tofile("dump_workspace.bin")
torch.npu.synchronize()
